pkg/tfbridge/validate_input_types_test.go

```diff
@@ -0,0 +1,2122 @@
+package tfbridge
+
+import (
+	"fmt"
+	"testing"
+
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestValidateInputTypesV2 tests the validateInputTypesV2 function
+func TestValidateInputType_objects(t *testing.T) {
+	testCases := []struct {
```

Comment:
Did we check MapType? Something like `Map<string,string>` or `Map<string,someObject>`. Skimming quickly not seeing it but perhaps need to look again.

Reply:
That should be what all of these tests are. `object` types.